src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1981,4 +1961,17 @@ private static function createRequestFromFactory(array $query = array(), array $
 
         return new static($query, $request, $attributes, $cookies, $files, $server, $content);
     }
+
+    public function setIpRetriever(IpRetrieverInterface $ipRetriever)
```

Comment:
Is it necessary? It would be better if we can keep `Request` as a value object. It sounds good to me to use a service to retrieve the "computed" IP but to have access to the raw data trough the `Request` object.